packages/nvidia_nat_core/src/nat/builder/function_info.py

```
@@ -604,18 +658,33 @@ def from_fn(fn: SingleCallableT | StreamCallableT,
+                            values = []
+
+                            no_args_stream_fn = typing.cast(Callable[[], AsyncGenerator[typing.Any]], stream_fn)
+                            async for m in no_args_stream_fn():
+                                values.append(m)
+
+                            return stream_arg.convert(values)
+
+                        single_fn = _stream_to_single_output_empty
+
+                    else:
+                        single_input_type = sig.parameters[list(sig.parameters.keys())[0]].annotation
```

Ruff (0.16.2)

[warning] Prefer `next(iter(sig.parameters.keys()))` over single element slice

Replace with `next(iter(sig.parameters.keys()))`

(RUF015)